tests/platforms/android/gradle/test_run.py

```diff
@@ -306,6 +362,94 @@ def mock_stream_output(app, stop_func, **kwargs):
     run_command.tools.mock_adb.kill.assert_not_called()
 
 
+def test_run_forward_reverse_ports(run_command, first_app_config):
+    """An app can be run with passthrough args."""
+    # Set up device selection to return a running physical device.
+    run_command.tools.android_sdk.select_target_device = mock.MagicMock(
+        return_value=("exampleDevice", "ExampleDevice", None)
+    )
+
+    # Set up the log streamer to return a known stream
+    log_popen = mock.MagicMock()
+    run_command.tools.mock_adb.logcat.return_value = log_popen
+
+    # To satisfy coverage, the stop function must be invoked at least once
+    # when invoking stream_output.
+    def mock_stream_output(app, stop_func, **kwargs):
+        stop_func()
+
+    run_command._stream_app_logs.side_effect = mock_stream_output
+
+    # Set up app config to have a `-` in the `bundle`, to ensure it gets
+    # normalized into a `_` via `package_name`.
+    first_app_config.bundle = "com.ex-ample"
+
+    # Invoke run_app with args.
+    run_command.run_app(
+        first_app_config,
+        device_or_avd="exampleDevice",
+        passthrough=[],
+        forward_ports=[80, 81],
+        reverse_ports=[78, 79],
+    )
+
+    # select_target_device was invoked with a specific device
+    run_command.tools.android_sdk.select_target_device.assert_called_once_with(
+        "exampleDevice"
+    )
+
+    # The ADB wrapper is created
+    run_command.tools.android_sdk.adb.assert_called_once_with(device="exampleDevice")
+
+    # The adb wrapper is invoked with the expected arguments
+    run_command.tools.mock_adb.install_apk.assert_called_once_with(
+        run_command.binary_path(first_app_config)
+    )
+    run_command.tools.mock_adb.force_stop_app.assert_called_once_with(
+        f"{first_app_config.package_name}.{first_app_config.module_name}",
+    )
+
+    assert run_command.tools.mock_adb.forward.call_count == 2
+    run_command.tools.mock_adb.forward.assert_any_call(80, 80)
+    run_command.tools.mock_adb.forward.assert_any_call(81, 81)
```

Comment:
Each of these 3-line groups could be replaced by a single assertion on [`mock_calls`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.mock_calls), which would give a much more useful message if it fails.